### PR TITLE
Use a stacklevel of 2 by default for all warnings

### LINKED ISSUE #1910: Refactor `warn` to use `stacklevel=2`
Probably refactor all instances of `warn` to instead use ``skimage._shared.warn`` and then implement it to always use `stacklevel=2`.

Also see http://permalink.gmane.org/gmane.comp.python.numeric.general/62455

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #1910.

### CODE REVIEW

File: skimage/__init__.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def _raise_build_error(e):
 
 
 if sys.version.startswith('2.6'):
-    warnings.warn("Python 2.6 is deprecated and will not be supported in scikit-image 0.13+")
+    warnings.warn("Python 2.6 is deprecated and will not be supported in scikit-image 0.13+", stacklevel=2)
```
Comment (ahojnnes) [inline skimage/__init__.py]: Wrap at 80.

File: skimage/measure/_ccomp.pyx [hunk 2]
```diff
@@ -49,7 +49,7 @@ cdef void get_bginfo(background_val, bginfo *ret) except *:
     if background_val is None:
         warnings.warn(DeprecationWarning(
```
Comment (ahojnnes) [inline skimage/measure/_ccomp.pyx]: Why note skimage._shared.warn?

File: skimage/viewer/utils/core.py [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
+from ..._shared.utils import warn
 import matplotlib as mpl
 from matplotlib.figure import Figure
 from matplotlib import _pylab_helpers
 from matplotlib.colors import LinearSegmentedColormap
 
 if has_qt and 'agg' not in mpl.get_backend().lower():
-    warnings.warn("Recommended matplotlib backend is `Agg` for full "
+    warn("Recommended matplotlib backend is `Agg` for full "
                   "skimage.viewer functionality.")
```
Comment (ahojnnes) [inline skimage/viewer/utils/core.py]: Indentation.

**Comment by ahojnnes** [comment]
A few comments, otherwise LGTM.

File: skimage/graph/_mcp.pyx [hunk 4]
```diff
@@ -304,7 +304,7 @@ cdef class MCP:
             self.flat_costs = costs.astype(FLOAT_D, copy=False).ravel('F')
         except TypeError:
             self.flat_costs = costs.astype(FLOAT_D).flatten('F')
-            warnings.warn('Upgrading NumPy should decrease memory usage and increase speed.', Warning)
+            warnings.warn('Upgrading NumPy should decrease memory usage and increase speed.', Warning, stacklevel=2)
```
Comment (soupault) [inline skimage/graph/_mcp.pyx]: Could you wrap at column 80 here?

**Comment by blink1073** [comment]
Comments addressed, builds passing.

File: skimage/__init__.py [hunk 5]
```diff
@@ -158,7 +158,9 @@ def _raise_build_error(e):
 
 
 if sys.version.startswith('2.6'):
-    warnings.warn("Python 2.6 is deprecated and will not be supported in scikit-image 0.13+")
+    msg = ("Python 2.6 is deprecated and will not be supported in"
```
Comment (JDWarner) [inline skimage/__init__.py]: Single space after 'in' is needed; implicit line continuation doesn't add one.

**Comment by JDWarner** [comment]
One comment, then :+1: :+1:!

**Comment by blink1073** [comment]
Space added.

**Comment by soupault** [comment]
:+1:

**Comment by ahojnnes** [comment]
Thanks.

**Pull request merged**